src/sentry/api/endpoints/team_details.py

```diff
@@ -83,7 +88,12 @@ def put(self, request: Request, team) -> Response:
                                owners can set this value.
         :auth: required
         """
-        if team.org_role != request.data.get("orgRole"):
+        team_org_role = team.org_role
+        if team_org_role != request.data.get("orgRole"):
+            if not features.has("organizations:org-roles-for-teams", team.organization, actor=None):
+                # remove the org role, but other fields can still be set
+                del request.data["orgRole"]
```

Comment:
not sure if this should be allowed, or if it should just error

Reply:
Should be okay either way. Seems like it would be significant only if a user had the flag on and then had it turned off (and only then if they're scripting against this endpoint).